source/Calamari/Deployment/PackageRetention/Repositories/JsonJournalRepository.cs

```diff
@@ -16,16 +17,24 @@ public class JsonJournalRepository
     {
         Dictionary<PackageIdentity, JournalEntry> journalEntries;
         const string SemaphoreName = "Octopus.Calamari.PackageJournal";
+        internal const string DefaultJournalName = "PackageRetentionJournal.json";
 
         readonly ICalamariFileSystem fileSystem;
         readonly ISemaphoreFactory semaphoreFactory;
-        readonly string journalPath;
+        internal string JournalPath { get; }
```

Comment:
Thinking this isn't the best way to do this (but not sure on how else). I've added a unit test to check that `JournalPath` is being set correctly based on the environment variables available which means this needs to be accessible by the test.

The alternative would be to have a more integration-test style test, with saving the file and checking it exists. But it doesn't look like we have an existing mock of `ICalamariFileSystem` which has this capability (closest thing is `TestCalamariPhysicalFileSystem`).